urllib3/connectionpool.py

```diff
@@ -239,11 +239,15 @@ def _get_conn(self, timeout=None):
         # If this is a persistent connection, check if it got disconnected
         if conn and is_connection_dropped(conn):
             log.info("Resetting dropped connection: %s" % self.host)
-            conn.close()
-            if getattr(conn, 'auto_open', 1) == 0:
-                # This is a proxied connection that has been mutated by
-                # httplib._tunnel() and cannot be reused (since it would
-                # attempt to bypass the proxy)
+            try:
+                conn.close()
+                if getattr(conn, 'auto_open', 1) == 0:
+                    # This is a proxied connection that has been mutated by
+                    # httplib._tunnel() and cannot be reused (since it would
+                    # attempt to bypass the proxy)
+                    conn = None
+            except Exception as e:
```

Comment:
`OpenSSL.SSL.SysCallError` is what you get with pyOpenSSL 0.15+: this was enabled in
<https://github.com/pyca/pyopenssl/commit/bff1d1ae72e743e41254bb31a5b50b5b63a37b85> [(changelog)](https://github.com/pyca/pyopenssl/blob/0.15/ChangeLog#L25)

With pyOpenSSL 0.14 I got some generic `Error` whose exact class I'm not sure of.

In general I'd agree to minimise catch/discard of errors. However in this particular case, it seems reasonable to catch Exception.  We are specifically only closing a connection where the underlying socket is known to have closed (`is_connection_dropped(conn)`), and we know we are going to discard it and create a new one.